eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -34,75 +34,45 @@ import fr.acinq.eclair.{wire, _}
 import kamon.Kamon
 import scodec.bits.ByteVector
 
-import scala.concurrent.duration._
-import scala.util.Random
-
 /**
  * This actor represents a logical peer. There is one [[Peer]] per unique remote node id at all time.
  *
- * The [[Peer] actor is mostly in charge of managing channels, stays alive when the peer is disconnected, and only dies
+ * The [[Peer]] actor is mostly in charge of managing channels, stays alive when the peer is disconnected, and only dies
  * when there are no more channels.
  *
  * Everytime a new connection is established, it is sent to the [[Peer]] and replaces the previous one.
  *
  * Created by PM on 26/08/2016.
  */
-class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, router: ActorRef, watcher: ActorRef, relayer: ActorRef, paymentHandler: ActorRef, wallet: EclairWallet) extends FSMDiagnosticActorLogging[Peer.State, Peer.Data] {
+class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, switchboard: ActorRef, router: ActorRef, watcher: ActorRef, relayer: ActorRef, paymentHandler: ActorRef, wallet: EclairWallet) extends FSMDiagnosticActorLogging[Peer.State, Peer.Data] {
```

Comment:
Why do we need to add the switchboard as dependency? Can't we assume that `context.parent` will always be the switchboard? Or is it because it's more future-proof and easier to test (which would be a completely valid argument)?

Reply:
It's easier to test, I got bitten with the `context.parent` and preferred to make things simpler.